### ISSUE TITLE
Issue: GPS tracking of 15-min counts does not work

### ISSUE THREAD

**Issue by AnneEskildsen** [BiologicalRecordsCentre/ABLE#683]
### Describe the issue

Danish volunteer Jan Nielsen, id 355516, reports that GPS tracking during 15-min count does not work/stops. He is not able to add the location of the count manually. Due to lack of spatial data his 15-min counts cannot be uploaded. 

### Expected behaviour

location of 15-min count should be tracked automatically via gps. 

### App Version

v1.26.0 (346)

### When did this happen

april-may 2024

### Additional context

While trying to fix the problem, Jan has accidentally deleted 6-7 counts, that had not been uploaded. Can these counts be recovered somehow?

**Comment by AnneEskildsen** [comment]
Device: iPhone 13, iOS-version: 17.4.1

**Comment by AnneEskildsen** [comment]
I forgot to mention that Jan reports that the Danish characters æ, ø and å are sometimes not accessible in the app.

**Comment by kazlauskis** [comment]
We can't recover any surveys that haven't been uploaded. Has the person given the location permission to the app? This is something they can verify in the device Settings page by opening the ButterflyCount app there and checking the page looks like this:

![Screenshot 2024-05-22 at 15 06 24](https://github.com/BiologicalRecordsCentre/ABLE/assets/7383162/8d9f03fa-4538-4572-adc6-6e9cd08fadc2)

**Comment by AnneEskildsen** [comment]
Yes, he confirms that his settings match the ones shown above, so this is not the problem. 

I wonder if this issue is somehow linked to the one I raised earlier this year about the app not tracking and displaying the exact route of the 15-min count?

**Comment by kazlauskis** [comment]
@AnneEskildsen  I understand there are two issues here: 
1. GPS not tracking. Note that for surveys that have ended, that is, after 15 minutes, users can't use the GPS.
2. Can't draw the route manually. Can they provide a step by step guide what they're doing and what doesn't work? Ideally with some screenshots so I can reproduce the issue.


> I wonder if this issue is somehow linked to the one I raised earlier this year about the app not tracking and displaying the exact route of the 15-min count?

Which issue do you mean?

**Comment by ahospers** [comment]
I had this problem several times on Xcover5 1.26.0 and also could not draw. Nowadays i add the last succesfull track to this problem so i can upload teh data. 
IF a track is attached to two datasets or if the trackdate is not within the observatin data but BEFORE than i had this problem. I think it happend five times or so in april with v1.26.0 


Van: ahospers <ahospers@ >
Date: wo 1 mei 2024 om 06:21
Subject: i got this, how to solve
To: ebms <ebms@ >


Van: Andr  <ahospers@ >
Date: wo 29 mei 2024 om 21:34
Subject: Fwd: i got this, how to solve
To: Karolis  <karolis@ >

**Comment by ahospers** [comment]
Probbably this screenshot Screenshot_20240426_143312_ButterflyCount.jpg
![Screenshot_20240426_143312_ButterflyCount](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/6366a1e8-6714-4efe-a9e8-e051f0e214a2)

**Comment by kazlauskis** [comment]
>  also could not draw

@ahospers Can you provide a step-by-step guide on what you're doing and what doesn't work when drawing an area? Ideally with some screenshots so I can reproduce the issue.

**Comment by ahospers** [comment]
Well, 15min are passed, you want to press "Klaar" finihed and that this popup happens.
The drawing does not function at all, nothing works. It seems it is disabled.
Than i go to recent tracks and look for the recent track which is closesed by and select that one.
Now i can save the track and upload it.
Samsung XCover 5

You just look at the location screen and the buttons do not work. Very very very annoying, irritating (Heel heel erg vervelend en frustrerend als je je trak op wilt slaan). I never use this screen so i have no idea how that works.

`
. He is not able to add the location of the count manually`

**Comment by kazlauskis** [comment]
@ahospers Thanks. Can you tell us step by step what you do on the Location page that doesn't work and attach screenshots?

**Comment by ahospers** [comment]
At least i now hava a schreenshot of how the GPS works. 
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/142594c2-3529-474a-8693-edfc15ddfca0)


![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/cf234958-cd57-4bb0-a00d-9dae54f8f6f5)

**Comment by ahospers** [comment]
I only have this text from the problem: 

One route  was  malfunctoining and  after opening the GPS, the GPS did not function,  i never succed in creating my own geography to leave the loop. It was just not possible to create an area.  For a normal user i t is hard to get rid of the route and leave the GPS and upload the track without GPStrack or with an created Geography.

**Comment by ahospers** [comment]
Today i had the issue again. It happens when yuou can not find ButterflyCouint as running app and you start ButterflyCount again and ou start anothter 15min Track. When you are finished you will find 2 concept tracks and in the drawing i see a zandloper conitinual running (=HourGlas). 
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/e44ad952-4039-4bcd-8553-a70a475e7353)
The running HourGlass wich took a lot of time . It seems the app is broken, vastgelopen.
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/1b0bc5e2-8cd0-4d43-acc1-3b70d9446fac)

**Comment by ahospers** [comment]
It takes a lot of time but sometimes i have this and i use the upper button to select another trac, 
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/0f1d3e24-ff7b-4387-8693-4bfc27404da4)
So i use the upper star button and i select an track from below and than i DO am able to press Klaar, Ready and after that UPLOAD button does apperar 
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/257dfccb-4185-4d89-84ad-f7bbec65b891)

**Comment by kazlauskis** [comment]
@ahospers I understand there are two issues: a) GPS isn't always tracking b) can't edit your track. So let's focus on the b) for now:

To edit a track:

1. Stop the GPS, toggle in the top-right corner of the Location page
2. Press Edit button with the pencil icon 
![Screenshot 2024-06-03 at 08 54 09](https://github.com/BiologicalRecordsCentre/ABLE/assets/7383162/c76d410a-701c-4ba8-8fe6-4cb81f995bf1)
4. Use the delete button with the bin icon to delete your invalid track
![Screenshot 2024-06-03 at 08 50 26](https://github.com/BiologicalRecordsCentre/ABLE/assets/7383162/10f889fa-dc37-41d5-a428-6072ab98d228)
6. Draw a new area or a trail

![Screenshot 2024-06-03 at 08 55 44](https://github.com/BiologicalRecordsCentre/ABLE/assets/7383162/2c4d3c56-a839-4121-a7e7-c701d9ba3a68)

**Comment by ahospers** [comment]
You can not use the top GPS toggle button, it wil go back in the original situation within a second.
 I  think b only occurs after a) has happend.
It seems to happen after double starting BUtterflyCount

**Comment by kazlauskis** [comment]
@ahospers are you saying the GPS toggle is always ON (green colour)?

**Comment by ahospers** [comment]
i think it is always off, but i was never been able to CHANGE it. so i think it is always grey and it flips immediately back to grey if you want  tot change it. Probably because the track is already finished ??? SO no GPS input is neccessary ??
I am only there after 15 minutes, the track/count has ended and i am NOT able to upload my track/15min count. I only do 15 minutes counts.

I do not think it is very inituif (logisch, vanzelfsprekend, handig, voor de hand liggend)  to throw away your current track.. but i hope i remember next time.

"Use the delete button with the bin icon to delete your invalid track"

**Comment by kazlauskis** [comment]
After 15 minutes, the GPS stops, and the toggle is disabled. Until then, it should be ON (green). If it isn't, then this is the problem a) we've discussed.

For problem b), track editing, the toggle is OFF, so you should be able to remove the track with the instructions I've sent you before and create a new one. Let me know if you can't draw a new track and, if so, what steps don't work for you.

**Comment by ahospers** [comment]
The situation is always after 15minutes, never before. One does not expect to throw something away as i presumes there is nothing, so that seems very onlogical. but i will try nextime.
The error also says that there is no Geographic information available.
I think this issue happens as one can not find ButterflyCOunt (my current android is very very very slow) and i start BC again and start a new track. In the end i have 2 concepts
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/62816285/6477e214-7025-47ec-9086-60d077e44a51)

**Comment by ahospers** [comment]
When i was in Poland it happend again (i was not able to upload the track) so i removed the butterflycount app waiting on a solved solution. If one can not upload the track you have a huge problem with the ButterflyCount app. In my opion the cause (oorzaak) is because it is possible to start ButterflyCOunt again while there is alreayd a butterflycount in the background running. This Butterflycount app is on Android not vissible but you can find out if there are two CONCEPT tracks, trails, lists. Maybe a warning that an already open, active track/trail/list is present could help so you will not start a new track but use the already existing, activ, running track. Does the app interfere with iNaturalist, Obsmap and OBsidentify as it is not visible sometimes?

**Comment by chrisvanswaay** [comment]
@kazlauskis 
I can confirm this has happened to me as well: you start a count, but sometimes (I have no idea why) the app jumps back to the startpage, where you might think you didn't push the + starbutton well enough, and then you do that again and a second parallel count starts. I now know I always have to check if there is a concept count first, but I have made this mistake in the past as well, very annoying. A check before starting a new count if there is an unfinished count still open, could prevent this.

**Comment by ahospers** [comment]
I thought it was more common since i have some demo, try, test version. Further it is always hard to find my data on this page https://butterfly-monitoring.net/elastic/all-records in my opiion it should just show up when you are logged in (inaturalist changes fully when you login in stead observation.org which hardly changes ) 
espescially when i want to change of check data it takes me too long to find the correct data on this website https://butterfly-monitoring.net/elastic/all-records

**Comment by ahospers** [comment]
I have something weird. After not able to upload a 15minute track in DDR or Poland i removed ButterflyCount app from Samsung XCover 5 Android. Today 1augu 2024 in Groningen Netherlands  i wanted to reinstall it but GooglePlay  gave the option to REACTIVATE it. I just did and i swa one CONCEPTTRACK which i opend with locations i UK England. Version 1.26.5(352)  
![image](https://github.com/user-attachments/assets/70dec639-9381-43ce-9a72-1b9557d70c48) 
![image](https://github.com/user-attachments/assets/0dcc9e56-c503-4de7-8b0f-ef08e2dce1a6)
![image](https://github.com/user-attachments/assets/b5857e65-3072-41cd-8776-1f737e4a229b)
![image](https://github.com/user-attachments/assets/15bfeaa3-c331-446b-afca-3d20b5134927)

![image](https://github.com/user-attachments/assets/afeb4c7b-8597-49ba-b69d-c2d882e81130)
![image](https://github.com/user-attachments/assets/a3819fd0-7d0e-4b5b-865d-09666a7f51ad)

**Comment by ahospers** [comment]
Last 5 years i have not been in UK, United Kingdom and i was wondering if i have everyseen a red trinange symbol near "Gebied/Area" 
![image](https://github.com/user-attachments/assets/126b2125-b00e-4bc0-bf04-bf3e1d50b844)

I always have a hard point in finding my data on the website. I expect it to be under my acount Top Right but there is no menu option https://butterfly-monitoring.net/user/3040 or under "Mijn Meetnet" https://butterfly-monitoring.net/netherlands-bms or under "Mijn data" where i never can find it. 
https://butterfly-monitoring.net/mydata/samples is not logical as it does NOT contain any location and not owner so it is hard to recognise teh data.

**Comment by CrisSevilleja** [comment]
> I thought it was more common since i have some demo, try, test version. Further it is always hard to find my data on this page https://butterfly-monitoring.net/elastic/all-records in my opiion it should just show up when you are logged in (inaturalist changes fully when you login in stead observation.org which hardly changes ) espescially when i want to change of check data it takes me too long to find the correct data on this website https://butterfly-monitoring.net/elastic/all-records

Hi @ahospers, to see your data submitted to the eBMS website you have to go to another page, My samples [https://butterfly-monitoring.net/mydata/samples](https://butterfly-monitoring.net/mydata/samples) where all the data submitted, transects, 15min counts or moth traps are visualised there and you can enter to edit them. 

Regarding the other issues, I assigned Karolis, the app developer to check this issue. I think the coordinates appearing in the UK can be related to the database location in the UK but indeed it is strange.

**Comment by ahospers** [comment]
In the app it is not called "samples", there are no "ids"in the app, there is no location (country, province, island and place) visible  and timestasmps of the observations are not visible and it is never called "EBMS 15minute survey"

https://butterfly-monitoring.net/elastic/all-records would be much much easier as there is an location , a recorder, an observation but unfortunetaly not a timestamp. I would prefer a symbol "i" far above any other symbol what might mean a DELETE button as one just does not know. 

I would keep everything 1:1. I have often the idea that the "EBSM 15mintut survey" has 20 differnt names. I would never translate the app (wel de soortnamen van de soorten) as i only creates errors in stead of making things clear.

**Comment by CrisSevilleja** [comment]
I agree with adding the time for the surveys on the app to know which poll it is, especially for the days when you have several surveys. The best thing would be to have a system similar to the one on the website, a configuration where you can choose what you want to see in the submitted surveys (time, country, location...), but at the moment we don't have the resources to improve all these features on the app or the website and we are focusing on the most important and urgent functionalities.

**Comment by ahospers** [comment]
Mijn broer kon vandaag ook de oplossing niet vinden om handmatig een track aan te maken. ER missen een aantal gegevens nl dat er al een track bestaat, dat je dire track weg moet gooien en dat je pas een nieuwe track kunt maken als je de oude weg gegooid hebt.
Het is geen logische workflow.
  https://github.com/BiologicalRecordsCentre/ABLE/issues/683#issuecomment-2144337994

**Comment by ahospers** [comment]
Since a few weeks, Since a few months this is happeing frequently, i get out of the enter page, data entry page to some where else since summer 2024.

I `can confirm this has happened to me as well: you start a count, but sometimes (I have no idea why) the app jumps back to the startpage, where you might think you didn't push the + starbutton well enough, and then you do that again and a second parallel co` 

Further i was succesfully able to throw away a bad track and mannually create a new polygon. Further i have questions about lump species: Sovon LiveAtlas heeft lump soorten als Roodoogjuffer(bij wrn.nl blijkbaar Erythromma), groene kikker,Koolwitje spec , bruinrode/steenrode maar hoe vind je handig de tegenhangers bij Butterflycount? Bij Bimbus terrestis gaat dat weer handig maar voor koolwitjes en heidelibellen zou ik graag tips en trucs horen. Moet je hier voor de latijn namen gebruiken van Koolwitje spec dan Pieris ?, Roodoogjuffer spec, Bruinrode/steenrode ? en Sympetrum voor gewoon heidelibellen ?

**Comment by ahospers** [comment]
Dit gebeurt sinds maart 2024 echt irritant vaak: "ned to me as well: you start a count, but sometimes (I have no idea why) the app jumps back to the startpage, where you might think you didn't push the + starbutton wel", de app is sinds maart 2024 echt veel slechter geworden en het gebeurt enorm vaak dat ik tracks gewoon niet up kan loaden. Ik wacht dan een dag (voor mijn doen extreem lang, ik meestal load ik meteen up omniet in de war te komen met die tracks en tegenwoordig kan ik dan sommige tracks die eerst niet te uploaden waren, weer wel uploaden. Maar mijn Samsung Xcover5 is echt een beroerd toestel, dus daar kan het zeker ook aan liggen. Blijft bijzonder dat het een dag later het uploaden wel lukt
